string[] names
float32[] angles
float32 fractionMaxSpeed
---
bool res